plugins/module_utils/identity/keycloak/keycloak.py

```diff
@@ -579,21 +582,30 @@ def add_client_roles_by_id_composite_rolemapping(self, rid, roles_rep, realm="ma
                                       % (rid, realm, str(e)))
 
     def add_group_rolemapping(self, gid, cid, role_rep, realm="master"):
-        """ Fetch the composite role of a client in a specified goup on the Keycloak server.
+        """ Add role of a client in a specified group on the Keycloak server.
 
-        :param gid: ID of the group from which to obtain the rolemappings.
-        :param cid: ID of the client from which to obtain the rolemappings.
+        :param gid: ID of the group.
+        :param cid: ID of the client.
         :param role_rep: Representation of the role to assign.
-        :param realm: Realm from which to obtain the rolemappings.
+        :param realm: Realm to add the rolemappings to.
         :return: None.
         """
-        available_rolemappings_url = URL_CLIENT_GROUP_ROLEMAPPINGS.format(url=self.baseurl, realm=realm, id=gid, client=cid)
-        try:
-            open_url(available_rolemappings_url, method="POST", http_agent=self.http_agent, headers=self.restheaders, data=json.dumps(role_rep),
-                     validate_certs=self.validate_certs, timeout=self.connection_timeout)
-        except Exception as e:
-            self.module.fail_json(msg="Could not fetch available rolemappings for client %s in group %s, realm %s: %s"
-                                      % (cid, gid, realm, str(e)))
+        if cid is None:
+            rolemappings_url = URL_REALM_GROUP_ROLEMAPPINGS.format(url=self.baseurl, realm=realm, id=gid)
+            try:
+                open_url(rolemappings_url, method="POST", http_agent=self.http_agent, headers=self.restheaders, data=json.dumps(role_rep),
+                        validate_certs=self.validate_certs, timeout=self.connection_timeout)
+            except Exception as e:
+                self.module.fail_json(msg="Could not add rolemappings to group %s, realm %s: %s"
+                                        % (gid, realm, str(e)))
+        else:
+            rolemappings_url = URL_CLIENT_GROUP_ROLEMAPPINGS.format(url=self.baseurl, realm=realm, id=gid, client=cid)
```

Comment:
Since both banches are mostly identical, why not simply have the `rolemappings_url = ...` in the `if` and everything else below?

Reply:
They do look similar, but rolemapping_url is a REALM url in one, and a CLIENT url in the other.

I can make the variable different names to clear up the confusion.

Reply:
This has no impact since the remainder of the code is still an exact copy (up to that variable name and the exact formulation of the error message).